overlord/snapstate/snapstate.go

```diff
@@ -1216,42 +1216,63 @@ func resolveChannel(st *state.State, snapName, newChannel string, deviceCtx Devi
 var errRevisionSwitch = errors.New("cannot switch revision")
 
 func switchSummary(snap, chanFrom, chanTo, cohFrom, cohTo string) string {
-	switchChannel := chanFrom != chanTo && chanTo != ""
-	switchCohort := cohFrom != cohTo && cohTo != ""
-	switch {
-	case switchChannel && !switchCohort && chanFrom == "":
-		return fmt.Sprintf(i18n.G("Switch snap %q from no channel to %q"),
-			snap, chanTo)
-	case switchChannel && !switchCohort:
-		return fmt.Sprintf(i18n.G("Switch snap %q from channel %q to %q"),
-			snap, chanFrom, chanTo)
-	case switchCohort && !switchChannel && cohFrom == "":
-		return fmt.Sprintf(i18n.G("Switch snap %q from no cohort to %q"),
-			snap, strutil.ElliptRight(cohTo, 10))
-	case switchCohort && !switchChannel:
-		return fmt.Sprintf(i18n.G("Switch snap %q from cohort %q to %q"),
-			snap, strutil.ElliptRight(cohFrom, 10), strutil.ElliptRight(cohTo, 10))
-	case switchCohort && switchChannel && chanFrom == "" && cohFrom == "":
-		return fmt.Sprintf(i18n.G("Switch snap %q from no channel to %q and from no cohort to %q"),
-			snap, chanTo, strutil.ElliptRight(cohTo, 10),
-		)
-	case switchCohort && switchChannel && cohFrom == "":
-		return fmt.Sprintf(i18n.G("Switch snap %q from channel %q to %q and from no cohort to %q"),
-			snap, chanFrom, chanTo, strutil.ElliptRight(cohTo, 10),
-		)
-	case switchCohort && switchChannel && chanFrom == "":
-		return fmt.Sprintf(i18n.G("Switch snap %q from no channel to %q and from cohort %q to %q"),
-			snap, chanTo, strutil.ElliptRight(cohFrom, 10), strutil.ElliptRight(cohTo, 10),
-		)
-	case switchCohort && switchChannel:
+	if cohFrom != cohTo {
+		if cohTo == "" {
+			// leave cohort
+			if chanFrom == chanTo {
+				return fmt.Sprintf(i18n.G("Switch snap %q away from cohort %q"),
+					snap, strutil.ElliptRight(cohFrom, 10))
+			}
+			if chanFrom == "" {
+				return fmt.Sprintf(i18n.G("Switch snap %q from no channel to %q and away from cohort %q"),
```

Comment:
I would s/from no channel//  everywhere, it's really a buggy state (that we might have had) but in practice it meant "stable", I wouldn't underline it here.